src/store/index.js

```diff
@@ -1,112 +1,93 @@
+/** This file should follow the guidelines outlined at
+ * https://docs.vuestorefront.io/guide/vuex/vuex-conventions.html
+ *
+ * With the following exceptions:
+ * 1. mutations should be camel-cased, not upper-cased. (they aren't constants like in other Flux implementations.)
+ * 2. to reset a state to it's default value, use the mutation name: reset<state>
+ * 3. Since we are a list management application, we can use "list," but it must refer to an actual List object.
+ *    likewise, "lists" must refer to an orderable list of List objects.
+ */
+
 import Vue from 'vue';
 import Vuex from 'vuex';
-import { auth } from '../main'; // eslint-disable-line import/no-cycle
+import {
+  auth,
+  globalPreferences,
+  listsCollection,
+  stateGroupsCollection,
+} from '../firebase';
 
 Vue.use(Vuex);
 
-const testLists = [
-  {
-    title: 'Test One',
-    description: 'Some random description.',
-    bgColor: '#ff00ff',
-    listItems: [
-      {
-        text: 'Item 1',
-        state: 'Done',
-      },
-      {
-        text: 'Item 2',
-        state: 'Not Done',
-      },
-    ],
-  },
-  {
-    title: 'Test Two',
-    description: 'Some random description.',
-    bgColor: '#ff0000',
-    listItems: [
-      {
-        text: 'Item 1',
-        state: 'Done',
-      },
-      {
-        text: 'Item 2',
-        state: 'Not Done',
-      },
-    ],
-  },
-  {
-    title: 'Test Three',
-    description: 'Some random description.',
-    bgColor: '#ffff00',
-    listItems: [
-      {
-        text: 'Item 1',
-        state: 'Done',
-      },
-      {
-        text: 'Item 2',
-        state: 'Not Done',
-      },
-    ],
-  },
-  {
-    title: 'Test Four',
-    description: 'Some random description.',
-    bgColor: '#465362',
-    listItems: [
-      {
-        text: 'Item 1',
-        state: 'Done',
-      },
-      {
-        text: 'Item 2',
-        state: 'Not Done',
-      },
-    ],
-  },
-];
-
 const defaultState = {
-  user: {
+  currentUser: {
     avatar: '',
     displayName: '',
     email: '',
   },
-  itemStates: ['Done', 'Not Done'],
-  lists: testLists,
+  lists: [],
 };
 
-export default new Vuex.Store({
+const store = new Vuex.Store({
   state: {
-    user: null,
-    lists: testLists,
+    currentUser: null,
+    lists: null,
+    currentList: { },
+    stateGroups: [],
+    globalPreferences: {
```

Comment:
there needs to be a default global preferences set, in order to not block rendering of content.